src/Makefile.am

```diff
@@ -836,7 +836,11 @@ lib_LTLIBRARIES += $(LIBBITCOINKERNEL)
 libbitcoinkernel_la_LDFLAGS = $(AM_LDFLAGS) -no-undefined $(PTHREAD_FLAGS)
 libbitcoinkernel_la_LIBADD = $(LIBSECP256K1) leveldb/libleveldb.la leveldb/libmemenv.la $(LIBUNIVALUE) crc32c/libcrc32c.la crc32c/libcrc32c_sse42.la $(LIBBITCOINCRYPTO)
 libbitcoinkernel_la_CPPFLAGS = $(AM_CPPFLAGS) -I$(builddir)/obj -I$(srcdir)/secp256k1/include -DBUILD_BITCOIN_INTERNAL $(BOOST_CPPFLAGS) $(LEVELDB_CPPFLAGS) -I$(srcdir)/$(UNIVALUE_INCLUDE_DIR_INT)
-libbitcoinkernel_la_CXXFLAGS = $(AM_CXXFLAGS) $(PIE_FLAGS)
+
+# libbitcoinkernel requires default symbol visibility, explicitly specify that
+# here so that things still work even when user configures with
+#   --enable-reduce-exports
+libbitcoinkernel_la_CXXFLAGS = $(AM_CXXFLAGS) $(PIE_FLAGS) -fvisibility=default
```

Comment:
In commit "build: Put reduced exports flags in RE_CXXFLAGS" (a324a9582d69bbc96d656b65b1b55d343aadb506)

This functional change seems only tangentially related to the cleanups in the rest of this commit, and like it should be in it's own behavior-changing commit or a part of the earlier commit adding this line in the first place (commit 08f2be497ee4769ff9d29d010433bf52c45ec103 "build: Extract the libbitcoinkernel library")

Reply:
Squashed it into 0057c5096ec798324208485642039d69b4b6f518 and added context in commit message